src/System.Management.Automation/engine/debugger/debugger.cs

```diff
@@ -1204,16 +1211,36 @@ internal VariableBreakpoint AddVariableBreakpoint(VariableBreakpoint breakpoint)
             return breakpoint;
         }
 
-        internal Breakpoint NewVariableBreakpoint(string path, string variableName, VariableAccessMode accessMode, ScriptBlock action)
+        private void UpdateBreakpoints(FunctionContext functionContext)
         {
-            CheckForBreakpointSupport();
-            return AddVariableBreakpoint(new VariableBreakpoint(path, variableName, accessMode, action));
-        }
+            if (functionContext._breakPoints == null)
+            {
+                // This should be rare - setting a breakpoint inside a script, but debugger hadn't started.
+                SetupBreakpoints(functionContext);
+            }
+            else
+            {
+                // Check pending breakpoints to see if any apply to this script.
+                if (string.IsNullOrEmpty(functionContext._file))
+                {
+                    return;
+                }
 
-        internal Breakpoint NewVariableBreakpoint(string variableName, VariableAccessMode accessMode, ScriptBlock action)
-        {
-            CheckForBreakpointSupport();
-            return AddVariableBreakpoint(new VariableBreakpoint(null, variableName, accessMode, action));
+                bool havePendingBreakpoint = false;
+                foreach ((int breakpointId, LineBreakpoint item) in _pendingBreakpoints)
+                {
+                    if (item.IsScriptBreakpoint && item.Script.Equals(functionContext._file, StringComparison.OrdinalIgnoreCase))
+                    {
+                        havePendingBreakpoint = true;
+                        break;
+                    }
+                }
+
+                if (havePendingBreakpoint)
+                {
+                    SetPendingBreakpoints(functionContext);
```

Comment:
Can this moved to line L#1234 instead? Then we do not need `havePendingBreakpoint`.

Reply:
Sure. Will be in next commit.